src/main/java/com/google/cloud/tools/app/module/DeleteAction.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app.module;
+
+import com.google.cloud.tools.app.Action;
+import com.google.cloud.tools.app.GCloudExecutionException;
+import com.google.cloud.tools.app.Option;
+import com.google.cloud.tools.app.ProcessCaller;
+import com.google.cloud.tools.app.ProcessCaller.ProcessCallerFactory;
+import com.google.cloud.tools.app.ProcessCaller.Tool;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Deletes a version of one or more modules.
+ */
+public class DeleteAction extends Action {
+
+  private static ProcessCallerFactory processCallerFactory = ProcessCaller.getFactory();
+  private Collection<String> modules = UNSET_COLLECTION;
```

Comment:
null has a slightly different meaning than UNSET_*, in the sense that the user can provide null, but can't provide UNSET_*. They happen to have the same value, which simplifies the implementation since later, we just need to check for whether a value is null, from not being set or from being set as null by the user.

It also makes the code more readable.